src/communication/communication_channel.hpp

```diff
@@ -71,6 +71,9 @@ namespace cubcomm
       css_error_code recv (char *buffer, std::size_t &maxlen_in_recvlen_out);
       css_error_code send (const std::string &message);
       css_error_code send (const char *buffer, std::size_t length);
+
+      // WARN: these functions use an all-or-nothing approach; use only if the higher level logic
+      // is able to recover from such a failure (important especially under load)
```

Comment:
These functions (send_int at least) is used in other contexts (connect_to_page_server).
The handling there seems to be escalation (do we retry at higher level in such case ?).

What are the cons with the other approach (to refactor send_int, recv_int to use safe buffer send and recv) ?

Reply:
@razvanradulescu apparently this does not solve our issue; even with this approach the recurring crash experienced in tests ... re-occurs; still pondering as to the actual cause